fixes async operations plus tag support

egbertn opened the pull request:
Bonjour Joeaudette!
Hope you find this usefull. The current 3.0.0 cannot work unless a global mvc setting allow synchroneous IO is set to true.

egbertn committed:
fixes async operations plus tag support

egbertn commented on src/cloudscribe.MetaWeblog/Controllers/MetaWeblogController.cs:
```diff
@@ -71,33 +71,34 @@ public virtual async Task<IActionResult> Index()
                 using (HttpContext.Request.Body)
                 {
-                    postedXml = XDocument.Load(HttpContext.Request.Body);
+                    postedXml = await XDocument.LoadAsync(HttpContext.Request.Body, LoadOptions.None, default);
```
Load will fail. Async fails too. Need to set AllowAsynchroneousIO = true

egbertn commented on src/cloudscribe.MetaWeblog/Controllers/MetaWeblogController.cs:
```diff
@@ -111,10 +112,11 @@ public virtual async Task<IActionResult> Index()
-                    postedXml.Save(s, SaveOptions.None);
-                }
+                using var s = System.IO.File.Create(requestFileName);
+                //await postedXml.SaveAsync(s, SaveOptions.None, default);
```
required on .net 3.0

egbertn commented on src/cloudscribe.MetaWeblog/Controllers/MetaWeblogController.cs:
```diff
@@ -154,10 +156,11 @@ public virtual async Task<IActionResult> Index()
+                using var s = System.IO.File.Create(reseponseFileName);
+                // await resultXml.SaveAsync(s, SaveOptions.None, default);
+                var xmlWriter = XmlWriter.Create(s, settings: new XmlWriterSettings { Async = true });
+                await resultXml.WriteToAsync(xmlWriter, default);
```
async required for 3.0.0

egbertn commented on src/cloudscribe.MetaWeblog/IMetaWeblogService.cs:
```diff
@@ -51,7 +51,12 @@ string authorDisplayName
-        
+
+        Task<IEnumerable<TagStruct>> GetTags(
+            string blogId,
```
new feature was not yet implemented. Now it is.

egbertn commented on src/cloudscribe.MetaWeblog/Utils.cs:
```diff
@@ -1,85 +1,28 @@
     public static class Utils
     {
-        public static string GetDateTimeStringForFileName()
+        public static string GetDateTimeStringForFileName(bool includeMiliseconds = false)
```
could not leave this, I know, bad conduct. But I Tested output.  Its ok.

egbertn commented on src/cloudscribe.MetaWeblog/XmlResult.cs:
```diff
@@ -32,26 +34,28 @@ public override void ExecuteResult(ActionContext context)
-                return Task.FromResult(0);
+               var xmlWriter = XmlWriter.Create(context.HttpContext.Response.Body, settings: new XmlWriterSettings { Async = true });
+
+                await Xml.WriteToAsync(xmlWriter, default);
```
required for .net core 3.x

egbertn commented on src/cloudscribe.MetaWeblog/cloudscribe.MetaWeblog.csproj:
```diff
@@ -11,24 +11,18 @@
-  <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Options" Version="3.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="3.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.0.0" />
```
these references are not required anymore

joeaudette commented:
Bonjour @egbertn !

In order to consider accepting your pull request, we need a signed contributor agreement.
https://raw.githubusercontent.com/cloudscribe/cloudscribe/master/copyright-assignment.txt

Thank you,

Joe

egbertn committed:
feature undo. Plus really need to set AllowSynchronousIO = true.

egbertn commented:
Hi Joe,

???
If it is on github and having apache 2.0 license, why this mistrust?
If I contribute this cannot in any way lead to claims on a public repo.
Just ignore the pull request. Ok?

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: Joe Audette<mailto:notifications@github.com>
Sent: zaterdag 4 januari 2020 21:57
To: cloudscribe/cloudscribe.MetaWeblog<mailto:cloudscribe.MetaWeblog@noreply.github.com>
Cc: Egbert Nierop<mailto:egbert_nierop@outlook.com>; Mention<mailto:mention@noreply.github.com>
Subject: Re: [cloudscribe/cloudscribe.MetaWeblog] fixes async operations plus tag support (#7)


Bonjour @egbertn<https://github.com/egbertn> !

In order to consider accepting your pull request, we need a signed contributor agreement.
https://raw.githubusercontent.com/cloudscribe/cloudscribe/master/copyright-assignment.txt

Thank you,

Joe

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cloudscribe/cloudscribe.MetaWeblog/pull/7?email_source=notifications&email_token=AA2PPB4F3MQGG3NY4J4WICDQ4DZ4XA5CNFSM4KCX3BE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIDAGRY#issuecomment-570819399>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AA2PPBZTSDOMV2YLCI2SGKDQ4DZ4XANCNFSM4KCX3BEQ>.

joeaudette commented:
Hi Egbert,

It is not a matter of mistrust, it is a matter of keeping copyright ownership purity so that we keep our options open. We may want to give the copyrights to the .net foundation in the future, or we may want to change the license to MIT or we may want to dual license to offer our customers a commercial license. In order to do any of those things we need to continue to own the code 100%.

See also our contributing guide https://github.com/cloudscribe/cloudscribe.MetaWeblog/blob/master/CONTRIBUTING.md
https://github.com/cloudscribe/cloudscribe/blob/master/CONTRIBUTING.md

It would be the same thing if you make a pull request on any well run open source project, if you want to make a pull request to asp.net core you have to sign an agreement with Microsoft, we are not big like Microsoft but it is still just as important and valid for us to maintain full copyright ownership of our projects. Same thing if you want to contribute to any major open source project like MySql or the linux kernel, you have to sign an agreement to do that. It is not all just handled by github.

Best Regards,

Joe

joeaudette commented:
FYI, I've moved the code from this repo into the cloudscribe.SimpleContent repo and fixed the async IO issues. Going forward any new work on cloudscribe.Metaweblog will happen in the SimpleContent repo
https://github.com/cloudscribe/cloudscribe.SimpleContent

The reason for this is because that repo makes it much easier to debug the metaweblog code by using SimpleContent and Open Live Writer.

Eventually we will delete the code from this repo.

@egbertn if you change your mind and decide to sign the agreement, then you can make a pull request in the SimpleContent repo to add the tags support. Otherwise you are free to fork the code and use a forked version.

joeaudette commented:
I've also published an updated nuget with the IO fix https://www.nuget.org/packages/cloudscribe.MetaWeblog/3.0.2

egbertn commented:
Ok will do later.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________
From: Joe Audette <notifications@github.com>
Sent: Sunday, January 5, 2020 4:12:09 PM
To: cloudscribe/cloudscribe.MetaWeblog <cloudscribe.MetaWeblog@noreply.github.com>
Cc: Egbert Nierop <egbert_nierop@outlook.com>; Mention <mention@noreply.github.com>
Subject: Re: [cloudscribe/cloudscribe.MetaWeblog] fixes async operations plus tag support (#7)


FYI, I've moved the code from this repo into the cloudscribe.SimpleContent repo and fixed the async IO issues. Going forward any new work on cloudscribe.Metaweblog will happen in the SimpleContent repo
https://github.com/cloudscribe/cloudscribe.SimpleContent

The reason for this is because that repo makes it much easier to debug the metaweblog code by using SimpleContent and Open Live Writer.

Eventually we will delete the code from this repo.

@egbertn<https://github.com/egbertn> if you change your mind and decide to sign the agreement, then you can make a pull request in the SimpleContent repo to add the tags support. Otherwise you are free to fork the code and use a forked version.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cloudscribe/cloudscribe.MetaWeblog/pull/7?email_source=notifications&email_token=AA2PPBYFZEHHZ222W7UNQJLQ4H2ETA5CNFSM4KCX3BE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIDY6PA#issuecomment-570920764>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AA2PPB2ND2MNJL2NODO3FILQ4H2ETANCNFSM4KCX3BEQ>.
